### PR TITLE
[HttpClient] Minor fixes

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 4.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php [hunk 1]
```diff
@@ -576,11 +576,11 @@ public function testResolve()
         ]);
 
         $this->assertSame(200, $response->getStatusCode());
-        $this->assertSame(200, $client->request('GET', 'http://symfony.com:8057/')->getStatusCode());
+        $this->assertSame(200, $client->request('GET', 'http://symfony.com:8057/', ['timeout' => $timeout = 0.1])->getStatusCode());
```
Comment (fancyweb) [inline src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: That prevents from waiting 3 sec while 0.1 is enough.

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php [hunk 2]
```diff
@@ -643,7 +643,6 @@ public function testDestruct()
     {
         $client = $this->getHttpClient(__FUNCTION__);
 
-        $downloaded = 0;
```
Comment (fancyweb) [inline src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: Unused var

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -1481,7 +1481,7 @@ private function addHttpClientSection(ArrayNodeDefinition $rootNode)
                                         ->info('A comma separated list of hosts that do not require a proxy to be reached.')
                                     ->end()
                                     ->floatNode('timeout')
-                                        ->info('Defaults to "default_socket_timeout" ini parameter.')
+                                        ->info('The idle timeout, defaults to "default_socket_timeout" ini parameter.')
```
Comment (fancyweb) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: Forgotten in https://github.com/symfony/symfony/pull/32928

Comment (nicolas-grekas) [reply src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: s/to/to the/

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php [hunk 4]
```diff
@@ -576,11 +576,11 @@ public function testResolve()
         ]);
 
         $this->assertSame(200, $response->getStatusCode());
-        $this->assertSame(200, $client->request('GET', 'http://symfony.com:8057/')->getStatusCode());
+        $this->assertSame(200, $client->request('GET', 'http://symfony.com:8057/', ['timeout' => $timeout = 0.1])->getStatusCode());
 
         $response = null;
         $this->expectException(TransportExceptionInterface::class);
-        $client->request('GET', 'http://symfony.com:8057/', ['timeout' => 3]);
+        $client->request('GET', 'http://symfony.com:8057/', ['timeout' => $timeout]);
```
Comment (nicolas-grekas) [inline src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: no need for the variable
let's use 1 to be sure we don't test a slow DNS resolver?

Comment (fancyweb) [reply src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: The variable is to be sure that the value was high enough to succesfully resolve in the previous assertion. Logically, the second time should not take much longer (note that the value I put is 1000x the time needed (for safety)). What about $timeout * 2 to be sure ? 1 sec seems huge.

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: why have a timeout at all here?

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php (hunk 4 above)
Comment (nicolas-grekas) [reply src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: I'm not sure 1s is that a big issue :)

File: src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php (hunk 1 above)
Comment (fancyweb) [reply src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php]: Related to https://github.com/symfony/symfony/pull/32948#discussion_r310592311 but I will remove the timeout from here and set 1 sec for the other.

**Comment by nicolas-grekas** [comment]
Thank you @fancyweb.

**Pull request merged**